packages/core/server/admin-api/services/query-layer-decorator.ts

```diff
@@ -77,8 +77,12 @@ const decorator = (service: IDecoratedService) => ({
       return service.update.call(this, uid, entityId, opts);
     }
 
-    // Generate the new path.
-    const generatedPath = await getPluginService('urlPatternService').resolvePattern(uid, { ...opts.data, id: entityId });
+    // Manually fetch the entity that's being updated.
+    // We do this becuase not all it's data is present in opts.data.
+    const entity = await service.findOne.call(this, uid, entityId);
+
+    // Generate the path.
+    const generatedPath = await getPluginService('urlPatternService').resolvePattern(uid, { ...opts.data, ...entity });
```

Comment:
I think we have to switch around `...opts.data` and `...entity` here.
As `opts.data` is the new data which should overwrite the old data.

Reply:
Ahh that makes sense, however I am not sure what old data means here as this is being done just after creation. But I think what you are saying might be technically more correct.

Reply:
Done